osbuild/util/selinux.py

```diff
@@ -35,18 +35,26 @@ def config_get_policy(config: Dict[str, str]):
     return config.get('SELINUXTYPE', None)
 
 
-def setfiles(spec_file: str, root: str, *paths):
+def setfiles(spec_file: str, root: str, *paths, exclude_paths: Optional[List[str]] = None) -> None:
```

Comment:
I am not python expert but if this is normal to put such complex expressions in the function declaration?

Reply:
Yea, that's common; type declarations always get a bit hairy.

Reply:
So what this means is `exclude_paths` is optional parameter, but if it is presented shall be a list of strings. And `-> None` means function shall returns nothing?

Reply:
What it means is that `exclude_paths` can either be `None` or a list of `str`, since it's given a default value of `None` that makes it an optional argument.

The `-> None` means that the function returns `None`.